### DISCUSSION TITLE
Getting error while adding Event ID in Telegraf

### DISCUSSION THREAD

**Discussion by ghost** [telegraf/telegraf#1720] [Q&A] [answered]
Error: 2022-10-28T16:17:32Z E! [inputs.win_eventlog] Error in plugin: Windows Event Log subscription error: The specified channel path is invalid.

I am using Event ID with XPath Query and getting above error. Below is the configuration that I have tried and none of them works.

From my config file:

[[inputs.win_eventlog]]
  ## Telegraf should have Administrator permissions to subscribe for some Windows Events channels
  ## (System log, for example)

  ## LCID (Locale ID) for event rendering
  ## 1033 to force English language
  ## 0 to use default Windows locale
  # locale = 0

  ## Name of eventlog, used only if xpath_query is empty
  ## Example: "Application"
  # eventlog_name = ""

  xpath_query = "Event/System[EventID=50]"

  process_userdata = true
  process_eventdata = true
  separator = "_"
  only_first_line_of_message = true
  timestamp_from_event = true
  event_tags = ["Source", "EventID", "Level", "LevelText", "Task", "TaskText", "Opcode", "OpcodeText", "Keywords", "Channel", "Computer"]
  event_fields = ["*"]
  exclude_fields = ["TimeCreated", "Binary", "Data_Address*"]
  exclude_empty = ["*ActivityID", "UserID"]

Previous configs that I have tried:

[[inputs.win_eventlog]]
  ## Telegraf should have Administrator permissions to subscribe for some Windows Events channels
  ## (System log, for example)

  ## LCID (Locale ID) for event rendering
  ## 1033 to force English language
  ## 0 to use default Windows locale
  # locale = 0

  ## Name of eventlog, used only if xpath_query is empty
  ## Example: "Application"
  # eventlog_name = ""

  xpath_query = '''
    "Event/System[EventID=50]"
  '''

  xpath_query = '''
  <QueryList>
    <Query Id="0" Path="System">
      <Select Path="System">
        *[System[(EventID='140')]]
	  </Select>
    </Query>
    <Query Id="0" Path="System">
      <Select Path="System">
	    *[System[(EventID='7034')]]
	  </Select>
    </Query>
  </QueryList>
  '''
  
  
  xpath_query = '''
    "Event/System[EventID=50]"
	"Event/System[EventID=140]"
    "Event/System[EventID=1792]"
    "Event/System[EventID=1038]"
    "Event/System[EventID=1069]"
    "Event/System[EventID=1795]"
    "Event/System[EventID=1135]"
    "Event/System[EventID=7031]"
    "Event/System[EventID=7034]"
  '''

**Comment by ghost** [comment]
I am using Telegraf 1.21.2

**Comment by MKRhere** [comment, marked as answer]
Wrong Telegraf. This is a Telegram Bot API library. I think you wanted [this](https://www.influxdata.com/time-series-platform/telegraf/).

**Reply by ghost** [reply]
This is the same Telegraf GitHub that I used to get the telegraf configuration. I appreciate your response and I will check the link you provided.
